src/components/forms/textinput/textinput.tsx

```diff
@@ -1,53 +1,115 @@
 import React from 'react';
 import styled from 'styled-components';
 import { TextStyles } from '../../typography';
+import { theme } from '../../layout/theme';
 
-const StyledLabel = styled.span`
+const TextArea = styled.textarea<{ hasError?: boolean }>`
+  height: 160px;
+  width: 100%;
+  padding: 19px 16px;
+  resize: vertical;
+
+  ${TextStyles.textR}
+
+  background: ${theme.palette.background.leadbox};
+  border: 2px solid
+    ${(props) =>
+      props.hasError
+        ? theme.palette.text.errorMessage
+        : theme.palette.background.leadbox};
+`;
+
+const Input = styled.input<{ hasError?: boolean }>`
+  width: 320px;
+  height: 48px;
+
+  padding-left: 16px;
+  padding-right: 16px;
+
+  ${TextStyles.textR}
+
+  background: ${theme.palette.background.leadbox};
+  border: 2px solid
+    ${(props) =>
+      props.hasError
+        ? theme.palette.text.errorMessage
+        : theme.palette.background.leadbox};
+`;
+
+const Label = styled.label`
   ${TextStyles.textXS}
   display: block;
   margin-bottom: 4px;
 
-  color: #202840;
+  color: ${theme.palette.text.default};
 `;
 
 const StyledErrorMessage = styled.span`
   ${TextStyles.textXS}
+  font-weight: 700;
   display: block;
   margin-top: 4px;
 
-  font-weight: 700;
-  color: #ff0d35;
+  color: ${theme.palette.text.errorMessage};
 `;
 
-interface TextInputProps {
-  errorMessage: string;
-  label: string;
+export interface TextInputProps {
+  inputRef: any;
```

Comment:
That's the legacy part among others so you can replace it within the current forms? I think that approach is okay but I would have done it much different. Let the legacy form stay as is and create a fresh table of components and forms. Much more fun to work within that environment and you can still  cherry pick whatever you want to reuse.

What's your plan? Slowly refactoring what we have into a new shape?

Reply:
Yes, my original plan was to refactor the existing components step by step. Therefore `input` and `textarea` are combined in one component (like in the legacy project). I like your suggestion to develop the new form independently from the existing implementation.